ipld/resolver/src/service.rs

```diff
@@ -720,3 +724,54 @@ async fn read_blob(iroh: Iroh, hash: Hash, offset: u32, len: u32) -> anyhow::Res
     debug!("read blob {}: {:?}", hash, res);
     Ok(res)
 }
+
+async fn extract_blob_hash_and_size(
```

Comment:
ah thanks for catching this! so, now every blob has a corresponding blob that stores its "hash sequence" and the actual blob is retrieved by fetching the first hash from the sequence.